int32 height
int32 width
int32 channels
uint8[] data
